### PR TITLE
don't require some args for ScheduledEvent.edit

### LINKED ISSUE #9261: ScheduledEvent edit method requires location for external events
### Summary

The edit method requires a location and end_time to edit the name of external events

### Reproduction Steps

1. Create an external scheduled event
2. Try to update the name of the event

### Minimal Reproducible Code

```python
import discord
from discord.ext import commands
from discord.utils import utcnow
import json
from datetime import timedelta

with open('config.json', 'r') as f:
    config = json.load(f)
token = config['BOT_TOKEN']
intents = discord.Intents.none()
intents.guilds = True

bot = commands.Bot(command_prefix="!", case_insensitive=True, intents=intents)

@bot.event
async def on_ready():
    print("Logged in.")
    guild = bot.guilds[0]
    start_time = utcnow() + timedelta(600)
    end_time = utcnow() + timedelta(1200)
    external = discord.EntityType.external
    privacy = discord.PrivacyLevel.guild_only
    location = "https://example.com"
    event = await guild.create_scheduled_event(name="MWE", start_time=start_time, end_time=end_time, entity_type=external, privacy_level=privacy, location=location)
    await event.edit(name="Edit")
    #await event.edit(name="Edit", location=location, end_time=end_time)

bot.run(token)
```

### Expected Results

I expected the name of my event to be updated.

Per https://discord.com/developers/docs/resources/guild-scheduled-event#modify-guild-scheduled-event-json-params
the location and end time are only required if *changing* to an external event. This works correctly when calling the raw discord API directly.

### Actual Results

I got a TypeError location must be set for external events. (And if I set location, same thing for end_time.)

2023-02-25 12:40:31 WARNING  discord.ext.commands.bot Privileged message content intent is missing, commands may not work as expected.
2023-02-25 12:40:31 INFO     discord.client logging in using static token
2023-02-25 12:40:32 INFO     discord.gateway Shard ID None has connected to Gateway (Session ID: f9802665bab0c999b126bbd8ab516c4f).
Logged in.
2023-02-25 12:40:34 ERROR    discord.client Ignoring exception in on_ready
Traceback (most recent call last):
  File "\~/.local/lib/python3.10/site-packages/discord/client.py", line 409, in _run_event
    await coro(*args, **kwargs)
  File "[..]/source/mwe.py", line 25, in on_ready
    await event.edit(name="Edit")
  File "\~/.local/lib/python3.10/site-packages/discord/scheduled_event.py", line 443, in edit
    raise TypeError('location must be set when entity_type is external')
TypeError: location must be set when entity_type is external

### Intents

guilds

### System Information

- Python v3.10.6-final
- [discord.py](http://discord.py/) v2.1.0-final
- aiohttp v3.8.1
- system info: Linux 5.15.0-60-generic # 66-Ubuntu SMP Fri Jan 20 14:29:49 UTC 2023

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

I would like to avoid having to store the location when a user creates the guild event, because currently I need to do that and provide the location again when editing the name/time. (I don't think it's reasonable to ask users to provide the location again when they simply want to change the name/time.)

### PR DESCRIPTION
**Pull request by z03h**
## Summary

Should fix #9261 
## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Rapptz** [comment]
Please see here: https://github.com/Rapptz/discord.py/pull/9264#issuecomment-1445181109

**Comment by z03h** [comment]
if the user passes in an entity type, should it assume they are trying to change the type?

Not sure if it's possible to add overload based on current attributes, such as end_time only being required if you are changing to external and an end time is not already set

**Comment by Rapptz** [comment]
> if the user passes in an entity type, should it assume they are trying to change the type?

Yes.

> Not sure if it's possible to add overload based on current attributes

It is, you have to specify `Literal` as the set value. Look at how it's done in `CommandTree.get_command`

**Comment by z03h** [comment]
> It is, you have to specify Literal as the set value. Look at how it's done in CommandTree.get_command

What I mean by that is when the user does `event.edit(end_time=None)`, if the current entity_type is external it should error but voice/stage should not. Not sure if there's a good way to go about that one

**Comment by Rapptz** [comment]
There isn't

File: discord/scheduled_event.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 from __future__ import annotations
 
 from datetime import datetime
-from typing import TYPE_CHECKING, AsyncIterator, Dict, Optional, Union
+from typing import TYPE_CHECKING, AsyncIterator, Dict, Optional, Union, overload, Literal
```
Comment (Puncher1) [inline discord/scheduled_event.py:28]:
```suggestion
from typing import TYPE_CHECKING, AsyncIterator, Dict, Literal, Optional, Union, overload
```

**Pull request merged**